DOC Add a note about the involvement of the contributor in maintenance. (#20044)
scikit-learn/scikit-learn: doc/developers/contributing.rst

   functionality is useful in practice and, if possible, compare it to other
   methods available in scikit-learn.

10. New features have some maintenance overhead. It is expected that the person
    who submits new code does part of the maintenance for that code at least

>     initially. New features need to be illustrated with narrative

Reviewer: > at least initially

What would be the time frame?

Author: I don't think it is realistic to ask for a specific lapse of time, I'm assuming the good faith of the contributor here and that the review process will be able to highlight their good willing.

> 10. New features have some maintenance overhead. It is expected that the person
>     who submits new code does part of the maintenance for that code at least
>     initially. New features need to be illustrated with narrative

Reviewer 2: Minor suggestion but LGTM, thanks

```suggestion
10. New features have some maintenance overhead. We expect PR authors
    to take part in the maintenance for the code they submit, at least
    initially. New features need to be illustrated with narrative
```
